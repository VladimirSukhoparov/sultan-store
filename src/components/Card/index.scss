$white: #ffffff;
$gray: #3f4e65;
$black: #111111;
$orange: #ffc85e;

.card {
  width: 326px;
  height: 490px;
  background: $white;
  box-shadow: 0px 14px 39px -11px rgba(0, 0, 0, 0.06);
  border-radius: 10px;
  box-sizing: border-box;
  padding: 0 20px;
  padding-top: 42px;
  display: flex;
  flex-direction: column;
  align-items: center;

  &__vector {
    opacity: 0.3;
    margin-bottom: -2px;
  }

  &__size {
    font-weight: 400;
    font-size: 12px;
    line-height: 130%;
    color: $gray;
    margin-left: 5px;
  }

  &__brand {
    display: block;
    text-decoration: none;
    font-style: normal;
    font-weight: 700;
    font-size: 16px;
    line-height: 130%;
    color: $black;
    margin: 0;
    margin-top: 10px;
    &:hover {
      text-decoration: underline;
    }
  }

  &__name {
    font-weight: 500;
    margin-left: 5px;
  }

  &__subText {
    font-weight: 300;
    font-size: 14px;
    line-height: 150%;
    color: $gray;
    margin: 0;
    margin-bottom: 5px;
  }

  &__upText {
    font-weight: 500;
    margin-left: 5px;
  }

  &__priceBtn {
    display: flex;
    align-items: center;
    margin-bottom: 26px;
  }

  &__price {
    font-weight: 800;
    font-size: 16px;
    line-height: 130%;
    margin: 0;
    margin-right: 40px;
    box-sizing: border-box;
    white-space: nowrap;
  }

  &__btn {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    padding: 10px 19px;
    gap: 5px;

    text-decoration: none;
    color: $white;
    font-weight: 700;
    font-size: 10px;
    line-height: 12px;
    letter-spacing: 0.15em;

    background-color: $orange;
    border-radius: 80px;
    width: 153px;
    height: 45px;
    box-sizing: border-box;

    cursor: pointer;

    img {
      width: 23.72px;
      height: 16.88px;
      margin-left: 1px;
      margin-bottom: -2px;
    }
  }

  &__image {
    max-height: 194px;
    display: block;
  }

  &__body {
    min-height: 254px;

    display: flex;
    flex-direction: column;
    justify-content: space-between;
  }
}
