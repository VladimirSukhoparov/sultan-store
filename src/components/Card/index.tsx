import React, { useState } from "react";
import { NavLink } from "react-router-dom";
import { FC } from "react";

import { useLocalStorage } from "../../hooks/useLocalStorage";

import "./index.scss";

type cardProps = {
  itemProd: {
    image: string;
    name: string;
    type: string;
    size: string;
    barcode: number;
    manufacturer: string;
    brand: string;
    description: string;
    price: string;
    typeOfCare: string[];
  };
  setBasket: (basket) => void;
};

export const Card: FC<cardProps> = ({ itemProd, setBasket }) => {
  const { writeLS } = useLocalStorage();
  const [product] = useState(itemProd);

  const addBasket = (item) => {
    writeLS("basket", item);
    setBasket((prevState) => [...prevState, item]);
  };

  return (
    <div className="card">
      <img
        src={require(`/src/components${product.image}`).default}
        alt="image"
        className="card__image"
      />
      <div className="card__body">
        <div>
          {product.type === "volume" ? (
            <img
              src={require("../images/vector1.png").default}
              alt=""
              className="card__vector"
            />
          ) : (
            <img
              src={require("../images/vector2.png").default}
              alt=""
              className="card__vector"
            />
          )}
          <span className="card__size">{product.size}</span>
          <NavLink
            to={`${product.barcode}`}
            className="card__brand"
            state={product}
          >
            {product.brand}
            <span className="card__name">{product.name}</span>
          </NavLink>
        </div>
        <div>
          <p className="card__subText">
            <span>Штрихкод:</span>
            <span className="card__upText">{product.barcode}</span>
          </p>
          <p className="card__subText">
            <span>Производитель:</span>
            <span className="card__upText">{product.manufacturer}</span>
          </p>
          <p className="card__subText">
            <span>Брэнд:</span>
            <span className="card__upText">{product.brand}</span>
          </p>
        </div>
        <div className="card__priceBtn">
          <p className="card__price">{product.price}</p>
          <a
            onClick={() => {
              addBasket(product);
            }}
            className="card__btn"
          >
            В корзину
            <img
              src={require("../images/basket2.png").default}
              alt="inBasket"
            />
          </a>
        </div>
      </div>
    </div>
  );
};
