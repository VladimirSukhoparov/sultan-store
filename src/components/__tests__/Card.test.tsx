import React from "react";
import { screen, render, cleanup } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";

import { Card } from "../Card";

afterEach(() => {
  cleanup();
});

test("should render card component", () => {
  render(
    <BrowserRouter>
      <Card
        itemProd={{
          image: "/images/aos.png",
          name: "Гель для душа",
          type: "volume",
          size: "450 мл",
          barcode: 4604049097540,
          manufacturer: "AOS",
          brand: "AOS",
          description:
            "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam interdum ut justo, vestibulum sagittis iaculis iaculis. Quis mattis vulputate feugiat massa vestibulum duis. Faucibus consectetur aliquet sed pellentesque consequat consectetur congue mauris venenatis. Nunc elit, dignissim sed nulla ullamcorper enim, malesuada.",
          price: "38.76 ₸",
          typeOfCare: ["body care", "hand care"],
        }}
        setBasket={function (basket: any): void {}}
      />
    </BrowserRouter>
  );
  const cardElement = screen.getByTestId("card");
  expect(cardElement).toBeInTheDocument();
});

test("matches snapshot", () => {
  const card = renderer
    .create(
      <BrowserRouter>
        <Card
          itemProd={{
            image: "/images/aos.png",
            name: "Гель для душа",
            type: "volume",
            size: "450 мл",
            barcode: 4604049097540,
            manufacturer: "AOS",
            brand: "AOS",
            description:
              "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam interdum ut justo, vestibulum sagittis iaculis iaculis. Quis mattis vulputate feugiat massa vestibulum duis. Faucibus consectetur aliquet sed pellentesque consequat consectetur congue mauris venenatis. Nunc elit, dignissim sed nulla ullamcorper enim, malesuada.",
            price: "38.76 ₸",
            typeOfCare: ["body care", "hand care"],
          }}
          setBasket={function (basket: any): void {}}
        />
      </BrowserRouter>
    )
    .toJSON();
  expect(card).toMatchSnapshot();
});
