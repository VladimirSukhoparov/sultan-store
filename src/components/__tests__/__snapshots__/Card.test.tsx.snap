// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="card"
  data-testid="card"
>
  <img
    alt="image"
    className="card__image"
    src="../../images/images/aos.png"
  />
  <div
    className="card__body"
  >
    <div>
      <img
        alt=""
        className="card__vector"
        src="vector1.png"
      />
      <span
        className="card__size"
      >
        450 мл
      </span>
      <a
        className="card__brand"
        href="/4604049097540"
        onClick={[Function]}
      >
        AOS
        <span
          className="card__name"
        >
          Гель для душа
        </span>
      </a>
    </div>
    <div>
      <p
        className="card__subText"
      >
        <span>
          Штрихкод:
        </span>
        <span
          className="card__upText"
        >
          4604049097540
        </span>
      </p>
      <p
        className="card__subText"
      >
        <span>
          Производитель:
        </span>
        <span
          className="card__upText"
        >
          AOS
        </span>
      </p>
      <p
        className="card__subText"
      >
        <span>
          Брэнд:
        </span>
        <span
          className="card__upText"
        >
          AOS
        </span>
      </p>
    </div>
    <div
      className="card__priceBtn"
    >
      <p
        className="card__price"
      >
        38.76 ₸
      </p>
      <a
        className="card__btn"
        onClick={[Function]}
      >
        В корзину
        <img
          alt="inBasket"
          src="basket2.png"
        />
      </a>
    </div>
  </div>
</div>
`;
